httpcore5/src/main/java/org/apache/hc/core5/http/message/StatusLine.java

```diff
@@ -89,6 +99,65 @@ public ProtocolVersion getProtocolVersion() {
     public String getReasonPhrase() {
         return this.reasonPhrase;
     }
+    
+    /**
+     * An enumeration representing the class of status code. Family is used
+     * here since class is overloaded in Java.
+     */
+    public enum Family {
+        /**
+         * {@code 1xx} HTTP status codes.
+         */
+        INFORMATIONAL,
+    
+        /**
+         * {@code 2xx} HTTP status codes.
+         */
+        SUCCESSFUL,
+        
+        /**
+         * {@code 3xx} HTTP status codes.
+         */
+        REDIRECTION,
+        
+        /**
+         * {@code 4xx} HTTP status codes.
+         */
+        CLIENT_ERROR,
+        
+        /**
+         * {@code 5xx} HTTP status codes.
+         */
+        SERVER_ERROR,
+        
+        /**
+         * Other, unrecognized HTTP status codes.
+         */
+        OTHER;
+
+        /**
+         * Get the response status family for the status code.
+         *
+         * @param statusCode response status code to get the family for.
+         * @return family of the response status code.
+         */
+        public static Family familyOf(final int statusCode) {
```

Comment:
This should be `valueOf` of just `of`. The `family` is redundant.

Reply:
> https://tools.ietf.org/html/rfc7231#section-6 talks about: the class of response
> 
> Maybe we should reuse this wording.

They are using `family` as `class` is overloaded in Java. We can use a synonym to family/class though (group, order, subclass, category, kind, variety, set, serie, etc.).

Reply:
I will make the changes tomorrow morning -- it is 1:30 AM here. :)

Reply:
Yes "class" is the proper name but I think we should call this enum `StatusClass` and put it in its own file.

Reply:
I agree with Gary here. Sound reasoning.